Source/WarpX.cpp

```diff
@@ -1643,3 +1643,82 @@ WarpX::PicsarVersion ()
+
+            // extract tileboxes for which to loop
+            Box const& tbx = mfi.tilebox(Bfield[0]->ixType().toIntVect()); /* just define which grid type */
+            Box const& tby = mfi.tilebox(Bfield[1]->ixType().toIntVect());
+            Box const& tbz = mfi.tilebox(Bfield[2]->ixType().toIntVect());
+
+            // loop over cells and output fields
+            amrex::ParallelFor(tbx, tby, tbz,
+            [=] AMREX_GPU_DEVICE (int i, int j, int k){
+
+            if(i==2 && j==4 && k==4){
+            std::ofstream ofs1("./Mfield_xface_left.txt", std::ofstream::app);
+            amrex::Print(ofs1).SetPrecision(16) << time << " " << M_xface(i,j,k,0) << " " << M_xface(i,j,k,1) << " " << M_xface(i,j,k,2) << " "
+                                                                   << M_yface(i,j,k,0) << " " << M_yface(i,j,k,1) << " " << M_yface(i,j,k,2) << " "
+                                                                   << M_zface(i,j,k,0) << " " << M_zface(i,j,k,1) << " " << M_zface(i,j,k,2) << std::endl;
+            ofs1.close();
+
+            std::ofstream ofs1_2("./Bfield_xface_left.txt", std::ofstream::app);
+            amrex::Print(ofs1_2).SetPrecision(16) << time << " " << Bx(i,j,k) << " " << By(i,j,k) << " " << Bz(i,j,k) << " " << std::endl;
+            ofs1_2.close();
+
+            std::ofstream ofs1_3("./Efield_xface_left.txt", std::ofstream::app);
+            amrex::Print(ofs1_3).SetPrecision(16) << time << " " << Ex(i,j,k) << " " << Ey(i,j,k) << " " << Ez(i,j,k) << " " << std::endl;
+            ofs1_3.close();
+
+            }
+
+            if(i==6 && j==4 && k==4){
+            std::ofstream ofs2("./Mfield_xface_right.txt", std::ofstream::app);
+            amrex::Print(ofs2).SetPrecision(16) << time << " " << M_xface(i,j,k,0) << " " << M_xface(i,j,k,1) << " " << M_xface(i,j,k,2) << " "
+                                                                   << M_yface(i,j,k,0) << " " << M_yface(i,j,k,1) << " " << M_yface(i,j,k,2) << " "
+                                                                   << M_zface(i,j,k,0) << " " << M_zface(i,j,k,1) << " " << M_zface(i,j,k,2) << std::endl;
+            ofs2.close();
+
+            std::ofstream ofs2_2("./Bfield_xface_right.txt", std::ofstream::app);
+            amrex::Print(ofs2_2).SetPrecision(16) << time << " " << Bx(i,j,k) << " " << By(i,j,k) << " " << Bz(i,j,k) << " " << std::endl;
+            ofs2_2.close();
+
+            std::ofstream ofs2_3("./Efield_xface_right.txt", std::ofstream::app);
+            amrex::Print(ofs2_3).SetPrecision(16) << time << " " << Ex(i,j,k) << " " << Ey(i,j,k) << " " << Ez(i,j,k) << " " << std::endl;
+            ofs2_3.close();
+
+            }
+              },
+
+            [=] AMREX_GPU_DEVICE (int i, int j, int k){
+
+              },
+
+            [=] AMREX_GPU_DEVICE (int i, int j, int k){
+
+              });
+        }
+    }
+#endif
+
+
```

Comment:
We should discuss with Reva how to handle these types of debugging functions (in terms of adding them to our primary branch)